x-pack/legacy/plugins/siem/public/components/edit_data_provider/index.tsx

```diff
@@ -113,25 +113,25 @@ export const StatefulEditDataProvider = React.memo<Props>(
           (saveElements[0] as HTMLElement).focus();
         }
       }
-    }
+    };
 
-    function onFieldSelected(selectedField: EuiComboBoxOptionProps[]) {
+    const onFieldSelected = useCallback((selectedField: EuiComboBoxOptionProps[]) => {
       setUpdatedField(selectedField);
 
       focusInput();
-    }
+    }, []);
 
-    function onOperatorSelected(operatorSelected: EuiComboBoxOptionProps[]) {
+    const onOperatorSelected = useCallback((operatorSelected: EuiComboBoxOptionProps[]) => {
       setUpdatedOperator(operatorSelected);
 
       focusInput();
-    }
+    }, []);
 
-    function onValueChange(e: React.ChangeEvent<HTMLInputElement>) {
+    const onValueChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
       setUpdatedValue(e.target.value);
-    }
+    }, []);
 
-    function disableScrolling() {
+    const disableScrolling = () => {
```

Comment:
Same comment here (and for `enableScrolling`) as with `focusInput` above. Should these just be extracted as helpers?